tests/test_dojo.py

```diff
@@ -0,0 +1,180 @@
+
+	def test_rejects_invalid_person_name_non_strings(self):
+		self.assertEqual(self.dojo.add_person("fellow", 125, 58888), 'Names in strings only')
+
+	
+	def test_adds_fellow_succesfully(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.add_person("fellow", "johny", "bravo")
+		self.assertTrue("Johny Bravo" in [person.full_name for person in self.dojo.people])
+
+	def test_adds_fellow_succesfully_with_accomadation(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.create_room("Hunteress", "livingspace")
+		self.dojo.add_person("fellow", "John", "Ngravo", "Y")
+		self.assertTrue("John Ngravo" in [person.full_name for person in self.dojo.people])
+
+	def test_adds_fellow_and_has_received_accomodation(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.create_room("Hunteress", "livingspace")
+		self.dojo.add_person("fellow", "John", "Ngravo", "Y")
+		self.assertTrue(self.dojo.people[0].accomodation.room_name == "Hunteress")
+
+	def test_adds_fellow_and_staff_received_offices(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.add_person("fellow", "johny", "bravo")
+		self.dojo.add_person("staff", "John", "Ngravo")
+		self.assertTrue(self.dojo.people[0].office.room_name == "Hunter")
+		self.assertTrue(self.dojo.people[1].office.room_name == "Hunter")
+
+	def test_adds_staff_succesfully(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.add_person("staff", "John", "Ngravo")
+		self.assertTrue("John Ngravo" in [person.full_name for person in self.dojo.people])
+		
+	def test_raise_error_on_duplicate_persons(self):
+		self.dojo.create_room("Hunter", "Office")
+		self.dojo.add_person("staff", "John", "Ngravo")
+		self.assertEqual(self.dojo.add_person("staff", "John", "Ngravo"), "Person already exists")
+
+class TesstPrintFunctions(unittest.TestCase):
+	"""A class to test the print functionalities in v1"""
+
+	def setUp(self):
+		self.dojo = Dojo()
+
+	def test_print_names_of_people_in_room(self):
+		#create rooms and add persons
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "ernest", "achesa", "N")
+		self.dojo.add_person("staff", "george", "wanjala")
+		#snapshot to store sys.stdout
+		temp = sys.stdout
+		#capture will hold the output being printed
+		capture = StringIO()
+		sys.stdout = capture
+		#call functions
+		self.dojo.print_room("Nairobi")
+		#end functions
+		#revert to snapshot of sys.stdout
+		sys.stdout = temp
+		#the printed value that was captured
+		output = capture.getvalue()
+		heading = "\x1b[33m\nROOM NAME: NAIROBI\tROLE\n"
+		expected_output = heading + "\x1b[0m\nErnest Achesa \t\tFELLOW\nGeorge Wanjala \t\tSTAFF\n"
+		self.assertEqual(output, expected_output)
+
+	def test_print_allocations_on_screen(self):
+		#create rooms and add persons
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.create_room("Chetambe", "livingspace")
+		self.dojo.add_person("staff", "george", "wanjala", "Y")
```

Comment:
@Gchesta  help me understand what's going on here! Are you testing a staff member who wants accommodation actually gets a `living space chetambe`